### PR TITLE
Enhanced video support

### PR DESCRIPTION
**Pull request by wbreza**
- Refactored canvas / asset preview
- Added test coverage for all editor components
- Wired up save video frame assets
- Wired up previous/next tagged asset buttons

### CODE REVIEW

File: src/react/components/pages/editorPage/editorPage.tsx [hunk 1]
```diff
@@ -131,7 +116,7 @@ export default class EditorPage extends React.Component<IEditorPageProps, IEdito
                 })}
```
Comment (tmiller-msft) [inline src/react/components/pages/editorPage/editorPage.tsx]: This doesn't seem to be changed, but I get a warning about the lack of keys on the array just above this for keyboard binding?

File: src/react/components/common/assetPreview/assetPreview.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
+import React from "react";
+import { IAsset, AssetType } from "../../../../models/applicationState";
+import { strings } from "../../../../common/strings";
+import { ImageAsset } from "./imageAsset";
+import { VideoAsset } from "./videoAsset";
+
+export type ContentSource = HTMLImageElement | HTMLVideoElement;
+
+export interface IAssetProps {
```
Comment (tbarlow12) [inline src/react/components/common/assetPreview/assetPreview.tsx]: nit: JSDocs?

File: src/react/components/common/assetPreview/imageAsset.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import React from "react";
+import { IAssetProps } from "./assetPreview";
+
+export interface IImageProps extends IAssetProps, React.ClassAttributes<ImageAsset> {
+}
+
+export class ImageAsset extends React.Component<IImageProps> {
```
Comment (tbarlow12) [inline src/react/components/common/assetPreview/imageAsset.tsx]: nit: JSDocs?

File: src/react/components/common/assetPreview/imageAsset.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import React from "react";
+import { IAssetProps } from "./assetPreview";
+
+export interface IImageProps extends IAssetProps, React.ClassAttributes<ImageAsset> {
```
Comment (tbarlow12) [inline src/react/components/common/assetPreview/imageAsset.tsx]: Is this just to have for future use?

File: src/react/components/common/assetPreview/videoAsset.tsx [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+import {
+    Player, BigPlayButton, ControlBar, CurrentTimeDisplay,
+    TimeDivider, PlaybackRateMenuButton, VolumeMenuButton,
+} from "video-react";
+import { IAssetProps } from "./assetPreview";
+import { IAsset, AssetType, AssetState } from "../../../../models/applicationState";
+import { AssetService } from "../../../../services/assetService";
+import { CustomVideoPlayerButton } from "../../common/videoPlayer/customVideoPlayerButton";
+
+export interface IVideoAssetProps extends IAssetProps, React.Props<VideoAsset> {
```
Comment (tbarlow12) [inline src/react/components/common/assetPreview/videoAsset.tsx]: nit: JSDocs on exported interfaces/classes

File: src/react/components/common/assetPreview/videoAsset.tsx [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+            >
+                <BigPlayButton position="center" />
+                <ControlBar autoHide={false}>
+                    <CurrentTimeDisplay order={1.1} />
+                    <TimeDivider order={1.2} />
+                    <PlaybackRateMenuButton rates={[5, 2, 1, 0.5, 0.25]} order={7.1} />
+                    <VolumeMenuButton enabled order={7.2} />
+                    <CustomVideoPlayerButton order={8.1}
+                        accelerator="ArrowLeft"
+                        tooltip="Previous Tagged Frame"
```
Comment (tbarlow12) [inline src/react/components/common/assetPreview/videoAsset.tsx]: Localization

File: src/react/components/common/assetPreview/videoAsset.tsx [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+                        onClick={this.movePreviousTaggedFrame}>
+                        <i className="fas fa-caret-left fa-lg" />
+                    </CustomVideoPlayerButton>
+                    <CustomVideoPlayerButton order={8.2}
+                        accelerator="ArrowRight"
+                        tooltip="Next Tagged Frame"
```
Comment (tbarlow12) [inline src/react/components/common/assetPreview/videoAsset.tsx]: Localization

File: src/react/components/common/keyboardBinding/keyboardBinding.tsx [hunk 8]
```diff
@@ -13,10 +13,15 @@ export class KeyboardBinding extends React.Component<IKeyboardBindingProps> {
...
-            this.props.accelerator,
-            this.props.onKeyEvent);
+        if (this.context && this.context.keyboard) {
+            this.deregisterBinding = this.context.keyboard.addHandler(
+                this.props.keyEventType || KeyEventType.KeyDown,
+                this.props.accelerator,
+                this.props.onKeyEvent,
+            );
+        } else {
+            console.warn("Keyboard Mananger context cannot be found - Keyboard binding has NOT been set.");
```
Comment (tbarlow12) [inline src/react/components/common/keyboardBinding/keyboardBinding.tsx]: Localization

File: src/react/components/pages/editorPage/editorPage.tsx [hunk 9]
```diff
@@ -42,8 +44,10 @@ export interface IEditorPageProps extends RouteComponentProps, React.Props<Edito
 export interface IEditorPageState {
```
Comment (tbarlow12) [inline src/react/components/pages/editorPage/editorPage.tsx]: nit: add new members to JSDocs

File: src/react/components/common/assetPreview/imageAsset.tsx (hunk 4 above)
Comment (wbreza) [reply src/react/components/common/assetPreview/imageAsset.tsx]: It's in use, just doesn't have any of its own members... it implements both `IAssetProps` and regular react props.

File: src/react/components/common/assetPreview/imageAsset.tsx (hunk 3 above)
Comment (wbreza) [reply src/react/components/common/assetPreview/imageAsset.tsx]: Done

File: src/react/components/common/assetPreview/videoAsset.tsx (hunk 5 above)
Comment (wbreza) [reply src/react/components/common/assetPreview/videoAsset.tsx]: Done

File: src/react/components/common/assetPreview/videoAsset.tsx (hunk 6 above)
Comment (wbreza) [reply src/react/components/common/assetPreview/videoAsset.tsx]: Thanks for calling me out on all of this 👍

Comment (wbreza) [reply src/react/components/common/assetPreview/videoAsset.tsx]: done

File: src/react/components/common/assetPreview/videoAsset.tsx (hunk 7 above)
Comment (wbreza) [reply src/react/components/common/assetPreview/videoAsset.tsx]: done

File: src/react/components/common/keyboardBinding/keyboardBinding.tsx (hunk 8 above)
Comment (wbreza) [reply src/react/components/common/keyboardBinding/keyboardBinding.tsx]: This is more of a developer warning and warns the developer if they attempt to use the `KeyboardBinding` outside of a `KeyboardManager` component.

File: src/react/components/pages/editorPage/editorPage.tsx (hunk 1 above)
Comment (wbreza) [reply src/react/components/pages/editorPage/editorPage.tsx]: Fixed.

**Pull request merged**